### PR TITLE
Updates internal `assert_allclose` callsites in favor of `assert_close`

### PR DESCRIPTION
**Pull request by pmeier**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63554 deprecate dtype getters from `torch.testing` namespace
* **#61841 Updates internal `assert_allclose` callsites in favor of `assert_close`**

Redo of #60863.

Differential Revision: [D30408145](https://our.internmc.facebook.com/intern/diff/D30408145)

### CODE REVIEW

File: torch/testing/_deprecated.py [hunk 1]
```diff
@@ -29,3 +36,48 @@ def inner_wrapper(*args: Any, **kwargs: Any) -> Any:
...
+
+
+def _get_default_rtol_and_atol(actual: torch.Tensor, expected: torch.Tensor) -> Tuple[float, float]:
+    actual_rtol, actual_atol = _DTYPE_PRECISIONS.get(actual.dtype, (0.0, 0.0))
+    expected_rtol, expected_atol = _DTYPE_PRECISIONS.get(expected.dtype, (0.0, 0.0))
+    return max(actual_rtol, expected_rtol), max(actual_atol, expected_atol)
+
+
+# TODO: include better migration instructions.
+@warn_deprecated("Use torch.testing.assert_close instead.")
```
Comment (pmeier) [inline torch/testing/_deprecated.py:55]: @mruberry Instead of posting a full upgrade guide here, is it ok to link to the RFC in #61844?

**Comment by mruberry** [comment]
Should we wait until assert_close is stable before we deprecate things in favor of it?

**Comment by pmeier** [comment]
> Should we wait until assert_close is stable before we deprecate things in favor of it?

See https://github.com/pytorch/pytorch/pull/61840#issuecomment-884135747.

**Review by mruberry — APPROVED** [review]
Cool! I updated the title to more accurately reflect the current state of the PR

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30408145).

**Pull request closed without merging**